gridstatus/isone_api/isone_api.py

```diff
@@ -382,3 +339,193 @@ def get_dayahead_hourly_demand(
         df = df.drop_duplicates(subset=["BeginDate", "Location"], keep="first")
 
         return self._handle_demand(df, interval_minutes=60)
+
+    def _handle_demand(
+        self,
+        df: pd.DataFrame,
+        interval_minutes: int = 60,
+    ) -> pd.DataFrame:
+        """
+        Process demand DataFrame: convert types, rename columns, and add Interval End.
+
+        Args:
+            df (pd.DataFrame): Input DataFrame with demand data.
+            interval_minutes (int): Duration of each interval in minutes. Default is 60.
+
+        Returns:
+            pd.DataFrame: Processed DataFrame.
+        """
+        try:
+            # Try the standard pandas datetime conversion first
+            df["Interval Start"] = pd.to_datetime(df["BeginDate"])
+            df["Interval Start"] = df["Interval Start"].dt.tz_convert(
+                self.default_timezone,
+            )
+        except AttributeError:
+            # NOTE(kladar) consistently the above logic fails for DST conversion days.
+            # This catches those and fixes it.
+            # Data coming out looks good, no missed intervals and no duplicates.
+            log.warning("Standard datetime conversion failed. Using custom parsing.")
+            df["Interval Start"] = df["BeginDate"].apply(
+                self.parse_problematic_datetime,
+            )
+
+        df = df.sort_values("Interval Start")
+        df["Interval End"] = df["Interval Start"] + pd.Timedelta(
+            minutes=interval_minutes,
+        )
+
+        df["Load"] = pd.to_numeric(df["Load"], errors="coerce")
+        df["Location Id"] = pd.to_numeric(df["LocId"], errors="coerce")
+
+        log.info(
+            f"Processed demand data: {len(df)} entries from {df['Interval Start'].min()} to {df['Interval Start'].max()}",
+        )
+        return df[["Interval Start", "Interval End", "Location", "Location Id", "Load"]]
+
+    @support_date_range("DAY_START")
+    def get_hourly_load_forecast(
+        self,
+        date: str | pd.Timestamp = "latest",
+        end: str | pd.Timestamp | None = None,
+        horizons: Literal["latest", "all"] = "all",
+        verbose: bool = False,
+    ) -> pd.DataFrame:
+        """
+        Get the hourly load forecast data for specified locations and date range.
+
+        Args:
+            date (str): The start date for the data request. Use "latest" for most recent data.
+            end_date (str | None): The end date for the data request. Only used if date is not "latest".
+            horizons (Literal["latest", "all"]): The horizons for the data request. Options are "latest" or "all", defaults to "all".
```

Comment:
Could we add a little more explanation of what horizons does? Maybe just move the comment from the function body into the docstring.

Reply:
Yep sounds good, I moved that to the docstring and expanded the explanation. Hopefully a bit more clear now.

Reply:
Thanks.